### DISCUSSION TITLE
TypeError: children.join is not a function

### DISCUSSION THREAD

**Discussion by himadrinath** [vercel/next.js#17786] [Help] [answered]
everything working fine in v9.5.3 after update to v9.5.4 start getting this error message

```
 TypeError: children.join is not a function
    at C:\Users\user\Documents\Social\node_modules\next\dist\next-server\server\render.js:17:275
    at Array.map (<anonymous>)
    at renderDocument (C:\Users\user\Documents\Social\node_modules\next\dist\next-server\server\render.js:17:86)
    at renderToHTML (C:\Users\user\Documents\Social\node_modules\next\dist\next-server\server\render.js:52:103)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async C:\Users\user\Documents\Social\node_modules\next\dist\next-server\server\next-server.js:81:66
    at async C:\Users\user\Documents\Social\node_modules\next\dist\next-server\server\next-server.js:74:142
    at async DevServer.renderToHTMLWithComponents (C:\Users\user\Documents\Social\node_modules\next\dist\next-server\server\next-server.js:106:365)
    at async DevServer.renderToHTML (C:\Users\user\Documents\Social\node_modules\next\dist\next-server\server\next-server.js:107:255)
    at async DevServer.renderToHTML (C:\Users\user\Documents\Social\node_modules\next\dist\server\next-dev-server.js:34:578)
    at async DevServer.render (C:\Users\user\Documents\Social\node_modules\next\dist\next-server\server\next-server.js:55:236)
```

i have to idea from where my code this generating from

**Comment by jamesmosier** [comment, marked as answer]
Hi @himadrinath! Without seeing any code this is going to be a bit hard to hep with, but taking a guess, do you have a `<Head>` element that is wrapped in multiple components or multiple levels of React Fragments?

[From the docs:](https://nextjs.org/docs/api-reference/next/head)

> 
title, meta or any other elements (e.g. script) need to be contained as direct children of the Head element, or wrapped into maximum one level of <React.Fragment> or arrays—otherwise the tags won't be correctly picked up on client-side navigations.

**Reply by LuizHAP** [reply]
Ok, i understand the problem, but how i fix this? I try to change my head scripts into this:

```
<Head>
   <React.Fragment>
       <meta name="keywords" content={metaKeywords} />
       <meta name="viewport" content="width=device-width, initial-scale=1.0" />
       <meta charSet="utf-8" />
   </React.Fragment>
</Head>
```

Or

```
<Head>
   <>
       <meta name="keywords" content={metaKeywords} />
       <meta name="viewport" content="width=device-width, initial-scale=1.0" />
       <meta charSet="utf-8" />
   </>
</Head>
```

But didn't fix this error.

**Comment by himadrinath** [comment]
i found the problem and fix it. thanks

**Reply by mattmcegg** [reply]
Hi, I might suggest in the future commenting what fixed it and what your root cause was.

**Reply by LuizHAP** [reply]
how u fix this? @himadrinath

**Comment by mattmcegg** [comment]
For other people curious, this was being caused by a face book script after updating to the latest version of Next ("next": "^9.5.5")

**Comment by tomchentw** [comment]
Hi all, I've encountered the same issue and I checked the code diff of `v9.5.3...v9.5.4`. It's the ` packages/next/client/head-manager.ts` changes that cause this error. See

https://github.com/vercel/next.js/compare/v9.5.3...v9.5.4#diff-e1a039509a8d92ebb1293f4f75a966cc981134c180b26f4422ed898862856b02

The key difference I observed is:

> The children under `<Head>` should be only a single level React element.

Hence, for my project, I will need to adjust my GTM code as follows:

```js
@@ -14,15 +14,19 @@ const noIndexHeadEl = (
 export const gtmHeadEl = (
   <Head>
     {/* <!-- Google Tag Manager --> */}
-    <noscript>
-      <iframe
-        title="gtm"
-        src="https://www.googletagmanager.com/ns.html?id=GTM-XXXXXXX"
-        height="0"
-        width="0"
-        style={{ display: "none", visibility: "hidden" }}
-      />
-    </noscript>
+    <noscript
+      dangerouslySetInnerHTML={{
+        __html: `
+<iframe
+  title="gtm"
+  src="https://www.googletagmanager.com/ns.html?id=GTM-XXXXXXX"
+  height="0"
+  width="0"
+  style="display:none;visibility:hidden;"
+></iframe>
+`,
+      }}
+    />
     <script
       type="text/javascript"
       dangerouslySetInnerHTML={{
```

---

Update: I was checking the output and it seems it produces the wrong HTML?

![JPEG-1](https://user-images.githubusercontent.com/922234/97436195-8d1b6300-195c-11eb-8b73-ab992ee035f6.jpg)

**Reply by rafaelfagundes** [reply]
That's the expected behavior. As soon as you disable javascript, the noscript tag will output the correct HTML in the body tag.

**Reply by DeBraid** [reply]
Thanks, using `dangerouslySetInnerHTML` inside `<noscript>` element worked for me.

**Comment by LuizHAP** [comment]
Ok, i understand the problem, but how i fix this? I try to change my head scripts into this:

```
<Head>
   <React.Fragment>
       <meta name="keywords" content={metaKeywords} />
       <meta name="viewport" content="width=device-width, initial-scale=1.0" />
       <meta charSet="utf-8" />
   </React.Fragment>
</Head>
```

Or

```
<Head>
   <>
       <meta name="keywords" content={metaKeywords} />
       <meta name="viewport" content="width=device-width, initial-scale=1.0" />
       <meta charSet="utf-8" />
   </>
</Head>
```

But didn't fix this error.

**Reply by himadrinath** [reply]
this that all your head and where u using it is it _app.js or somewhere else

**Comment by chaddjohnson** [comment]
For anyone stuck on this, this helped me: https://github.com/vercel/next.js/discussions/16814#discussioncomment-61029.

**Reply by nghiepnguyen520** [reply]
I don't think it can help me with this case.

**Comment by mparramont** [comment]
In case it helps anybody, it was happening to me because I had a `<div>` inside my `<Head>`:
```js
      <Head>
        <title>
          <div>Some title</div>
        </title>
        <link rel="icon" href="/favicon.ico" />
      </Head>
```

This fixed it:
```js
      <Head>
        <title>Some title</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>
```
